js/src/controls/GeoportalControls.js

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+const L = require('../leaflet.js');
+const control = require('../controls/Control.js');
+
+
+export class LeafletGeoportalLayerSwitcherModel extends control.LeafletControlModel {
+  defaults() {
+    return {
+      ...super.defaults(),
+     _view_name: 'LeafletGeoportalLayerSwitcherView',
+      _model_name: 'LeafletGeoportalLayerSwitcherModel'
+      };
+    }
+  }
+
+  export class LeafletGeoportalLayerSwitcherView extends control.LeafletControlView {
+    initialize(parameters) {
+      super.initialize(parameters);
+      this.map_view = this.options.map_view;
+    }
+    create_obj() {
+      this.obj = L.geoportalControl.LayerSwitcher()
+    }
+  }
+
+  export class LeafletGeoportalSearchEngineModel extends control.LeafletControlModel {
+    defaults() {
+      return {
+        ...super.defaults(),
+        _view_name: 'LeafletGeoportalSearchEngineView',
+        _model_name: 'LeafletGeoportalSearchEngineModel',
+        position : 'topleft',
+        collapsed : true,
+        zoomTo : 'auto',
+        displayInfo : true,
+        apiKey : 'cartes',
+        displayAdvancedSearch : true,
+        resources : ["PositionOfInterest", "StreetAddress"],
+        advancedSearch : {},
+        geocodeOptions : {},
+        autocompleteOptions : {}
+        };
+      }
+    }
+
+    export class LeafletGeoportalSearchEngineView extends control.LeafletControlView {
+      initialize(parameters) {
+        super.initialize(parameters);
+        this.map_view = this.options.map_view;
+      }
+      create_obj() {
+        this.obj = L.geoportalControl.SearchEngine(this.get_options())
+      }
+    }
+
+    export class LeafletGeoportalRouteModel extends control.LeafletControlModel {
+      defaults() {
+        return {
+          ...super.defaults(),
+          _view_name: 'LeafletGeoportalRouteView',
+          _model_name: 'LeafletGeoportalRouteModel',
+          position : "bottomleft",
+          collapsed : true,
+          exclusions : {"toll" : true,
+                        "bridge" : false,
+                        "tunnel" : true
+                      },
+          graphs : ['Pieton', 'Voiture'],
+          autocompleteOptions : {},
+          routeOptions : {}
+          };
+        }
+      }
+
+      export class LeafletGeoportalRouteView extends control.LeafletControlView {
+        initialize(parameters) {
+          super.initialize(parameters);
+          this.map_view = this.options.map_view;
+        }
+        create_obj() {
+          this.obj = L.geoportalControl.Route(this.get_options())
```

Comment:
There is no option on the `GeoportalRoute` Python side?

Reply:
There are some options for `GeoportalRoute` in geoportal.py (from line 72 to 79).